SierraSoftworks/tailscale-udm
# feat: Add support for automatically installing Tailscale after a firmware upgrade

This PR adds experimental support for automatically installing Tailscale following a UniFi OS 2.x firmware upgrade (as well as updating Tailscale after a reboot, if an update is available). It does so by installing a `systemd` unit which runs before `tailscaled` in the startup sequence and triggers the `/data/manage.sh on-boot` command (in much the same way as the UniFi OS 1.x on_boot.d script worked).

I have not had an opportunity to test this on a full firmware upgrade, however indications are that `systemd` units persist across firmware upgrades, which should ensure that this works as expected.

## Testing Steps
To validate that this new release works correctly on your device, please follow these steps to install a prerelease version of it:

1. Download the latest release package (from the `Build / Package` pipeline - [here](https://github.com/SierraSoftworks/tailscale-udm/suites/11874801757/artifacts/621070788)) onto your device.
   ```sh
   scp ./package.zip root@unifi:/tmp/tailscale-udm-pre.zip
   ```
2. Unzip the build artifact package using `unzip -oq /tmp/tailscale-udm-pre.zip -d /tmp`
3. Un-tar the installation package into your `/data/tailscale` directory using `tar xzf /tmp/tailscale-udm.tgz -C /data/`
4. Run the force installation command: `/data/tailscale/manage.sh install!`
5. Confirm that the message `Installing pre-start script to install Tailscale on firmware updates.` is printed to your terminal.
6. Install the latest firmware update for your device using `ubnt-systool fwupdate <url-of-firmware-package>` and confirm that Tailscale is present and running after the firmware update completes. (Note that you can install the current firmware version if you wish to test immediately).

See #38 for further information.
